esphome/components/pylontech/__init__.py

```diff
@@ -0,0 +1,46 @@
+import logging
+import esphome.codegen as cg
+from esphome.components import uart
+import esphome.config_validation as cv
+from esphome.const import CONF_ID
+
+_LOGGER = logging.getLogger(__name__)
+
+CODEOWNERS = ["@functionpointer"]
+DEPENDENCIES = ["uart"]
+MULTI_CONF = True
+
+CONF_PYLONTECH_ID = "pylontech_id"
+CONF_BATTERY = "battery"
+
+pylontech_ns = cg.esphome_ns.namespace("pylontech")
+PylontechComponent = pylontech_ns.class_(
+    "PylontechComponent", cg.PollingComponent, uart.UARTDevice
+)
+PylontechBattery = pylontech_ns.class_("PylontechBattery")
+
+CV_NUM_BATTERIES = cv.int_range(1, 6)
+
+PYLONTECH_COMPONENT_SCHEMA = cv.Schema(
+    {
+        cv.Required(CONF_PYLONTECH_ID): cv.use_id(PylontechComponent),
```

Comment:
This can be used for automatic id resolving
```suggestion
        cv.GenerateID(CONF_PYLONTECH_ID): cv.use_id(PylontechComponent),
```